src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -153,10 +153,10 @@ private void consume() throws Exception {
         _sensors._consumeError.record();
         continue;
       }
-      int partition = record.partition();
-      long index = (Long) avroRecord.get(DefaultTopicSchema.INDEX_FIELD.name());
+      int partition = (int) avroRecord.get(DefaultTopicSchema.PARTITION_FIELD.name());
```

Comment:
So purely using the message payload to identify the partition will still cause the messages to come out-of-order. In the solution submitted here, the source cluster partition is passed through the message payload and also used as the message key. By doing this I am able to enforce order. After submitting this patch, I realized that passing the source cluster partition in the message payload as well as using it as the key is superfluous. We can simply use the source cluster partition as the key and then store expected indexes using the RecordMetadata key. While the currently proposed solution works fine, it can be more efficient and cleaner by simply using the RecordMetadata.